services/mail/mail_sender.go

```diff
@@ -1,28 +1,77 @@
 package mail
 
 import (
+	"context"
+	"encoding/base64"
 	"log"
-	"net/smtp"
 	"os"
+	"time"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/gmail/v1"
+	"google.golang.org/api/option"
 )
 
+//EMAIL_SECRET=C65v5DEIHlDdEU1shHYrlnns
+//EMAIL_CLIENT=918242223374-k6ihlaib00kvt65j82gnqnrukms6ocrg.apps.googleusercontent.com
+
 //send a text mail to the receiver's email id
 func SendMail(to string, subject string, body string) {
-	// Sender data.
-	from := os.Getenv("EMAIL_SERVER_USER")
-	password := os.Getenv("EMAIL_SERVER_PASS")
-	smtpHost := os.Getenv("EMAIL_SMTP_HOST")
-	smtpPort := os.Getenv("EMAIL_SMTP_PORT")
-
-	// Message.
-	msg := "From: Codephile" + "\n" + "To: " + to+ "\n" + "Subject: "+ subject + "\n\n" + body
-	// Authentication.
-	auth := smtp.PlainAuth("", from, password, smtpHost)
-
-	// Sending email.
-	err := smtp.SendMail(smtpHost + ":" + smtpPort, auth, from, []string{to}, []byte(msg))
+	var GmailService *gmail.Service
+
+	config := oauth2.Config{
+		ClientID:     os.Getenv("EMAIL_CLIENT"),
+		ClientSecret: os.Getenv("EMAIL_SECRET"),
+		Endpoint:     google.Endpoint,
+		RedirectURL:  os.Getenv("REDIRECT_URL"),
+	}
+
+	token := oauth2.Token{
+		AccessToken:  os.Getenv("EMAIL_ACCESS_TOKEN"),
+		RefreshToken: os.Getenv("EMAIL_REFRESH_TOKEN"),
+		TokenType:    "Bearer",
+		Expiry:       time.Now(),
+	}
+
+	//refreshing token
+	var tokenSource = config.TokenSource(context.TODO(), &token)
+
+	updatedToken, Error := tokenSource.Token()
+	if Error != nil {
+		log.Fatalf(Error.Error())
+	} else if (*updatedToken).AccessToken != token.AccessToken {
+		os.Setenv("EMAIL_ACCESS_TOKEN", (*updatedToken).AccessToken)
```

Comment:
We can add a local variable as we have done it for other services.  This would anyway be lost when the server process is killed.